cfme/cloud/flavor.py

```diff
@@ -108,11 +143,66 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=True)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            self.wait_for_disappear()
+
+    def wait_for_disappear(self):
+        """Wait for flavor to disappear"""
+        wait_for(self.provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=300)
```

Comment:
Question: Why are you waiting for the provider to refresh here? Would just waiting for the Flavor object to not present in the table not be enough?